modules/input/react/lib/Input.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import styled from 'react-emotion';
+import {GrowthBehavior, ErrorType, errorRing} from '@workday/canvas-kit-react-common';
+import {inputColors, spacingNumbers, type} from '@workday/canvas-kit-react-core';
+
+export interface InputProps extends GrowthBehavior, React.InputHTMLAttributes<HTMLInputElement> {
+  disabled?: boolean;
+  error?: ErrorType;
+  hasIcon?: boolean;
+  inputRef?: React.Ref<HTMLInputElement>;
+  onChange?: React.ChangeEventHandler<HTMLInputElement>;
+  placeholder?: string;
+  readOnly?: boolean;
+  type?: string;
+  value?: any;
```

Comment:
I'm surprised we don't disallow `any`. It is a bit more work, but would a generic type be more appropriate? Otherwise `HTMLInputElement` types `value` as a `string` since the input element can only represent strings. 

Supporting other types as a higher level concept would require serializing and deserializing to satisfy the `<input>` element. And serializing/deserializing requires care to not change what the user is typing while typing. Ex: a DateInput wrapping this input serialize/deserialize to a `Date` object not changing the characters representing the date while the user is interacting. One could imagine the frustration. That might require a whole lifecycle subsystem of parsing, validation and serializing to get right.

I don't have much context here. How is this handled elsewhere?

Reply:
@anicholls @lychyi Do you have ideas or past experience with this before?

I've done low-level inputs that represented higher-level types like `number`, `Date` and other 2-way transformative types in the past. They are tricky, but doable. Best to solve at the lowest layer possible. AngularJS 1.x was originally created just to tackle forms (validation, parsing, etc)